packages/third-party-notices/src/write-third-party-notices.ts

```diff
@@ -64,6 +65,27 @@ export async function writeThirdPartyNotices(
   }
 }
 
+export async function writeThirdPartyNoticesFromMap(
+  options: WriteThirdPartyNoticesOptions,
+  moduleNameToPathMap: Map<string, string>
+): Promise<void> {
+  const { additionalText, json, preambleText, sourceMapFile } = options;
+  let { outputFile } = options;
+
+  const licenses = await extractLicenses(moduleNameToPathMap);
+  const outputText = json
+    ? createLicenseJSON(licenses)
+    : createLicenseFileContents(licenses, preambleText, additionalText);
+
+  if (!outputFile) {
+    const prefix = sourceMapFile?.replace(/\.map$/, "") ?? "main";
+    outputFile = prefix + ".LICENSE.txt";
+  }
+
+  const writeFileAsync = promisify(fs.writeFile);
+  await writeFileAsync(outputFile, outputText, "utf8");
```

Comment:
Can we use `fs.writeFileSync` here? It's going to be faster not to wait for a tick in Node.
```suggestion
  fs.writeFileSync(outputFile, outputText);
```

Reply:
Fixed.